core/sql/ustat/hs_globals.cpp

```diff
@@ -2887,6 +2890,22 @@ Lng32 HSGlobalsClass::Initialize()
         HSHandleError(retcode);
         TM->Commit(); // Must commit this transaction (even if schema didn't get created).
       }
+                                              /*=====================================*/
+                                              /*   CREATE HBASE STATS SCHEMA         */
+                                              /*   typically as trafodion.hbasestats */
+                                              /*=====================================*/
+    if (isNativeHbaseCat(objDef->getCatName()))
+      {
+        HSTranMan *TM = HSTranMan::Instance(); // Must have transaction around this.
+        TM->Begin("Create schema for native hbase stats.");
```

Comment:
Yes. the Begin() and Commit() methods detect any existing transactions in
progress and if one if found, these two methods are no op.

On Mon, Sep 21, 2015 at 11:05 AM, DaveBirdsall <notifications@github.com>
wrote:

> In core/sql/ustat/hs_globals.cpp
> <https://github.com/apache/incubator-trafodion/pull/87#discussion_r39989234>
> :
>
> > @@ -2887,6 +2890,22 @@ Lng32 HSGlobalsClass::Initialize()
> >          HSHandleError(retcode);
> >          TM->Commit(); // Must commit this transaction (even if schema didn't get created).
> >        }
> > +                                              /*=====================================*/
> > +                                              /*   CREATE HBASE STATS SCHEMA         */
> > +                                              /*   typically as trafodion.hbasestats */
> > +                                              /*=====================================*/
> > +    if (isNativeHbaseCat(objDef->getCatName()))
> > +      {
> > +        HSTranMan *TM = HSTranMan::Instance(); // Must have transaction around this.
> > +        TM->Begin("Create schema for native hbase stats.");
>
> Is it known that there is no user transaction active in this code path?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/87/files#r39989234>.
>



-- 
Regards, --Qifan